main.go

```diff
@@ -137,3 +140,62 @@ func registerCapabilities(port string, log logger.Handler) {
 		log.Info(err.Error())
 	}
 }
+
+func registerDynamicCapabilities(port string, log logger.Handler) {
+	registerWorkloads(port, log)
+	//Start the ticker
+	const reRegisterAfter = 24
+	ticker := time.NewTicker(reRegisterAfter * time.Hour)
+	for {
+		<-ticker.C
+		registerWorkloads(port, log)
+	}
+
+}
+
+func registerWorkloads(port string, log logger.Handler) {
+	appVersion, chartVersion, err := getLatestValidAppVersionAndChartVersion()
+	if err != nil {
+		log.Info("Could not get latest version")
+		return
+	}
+	log.Info("Registering latest workload components for version ", appVersion)
+	// Register workloads
+	if err := adapter.RegisterWorkLoadsDynamically(mesheryServerAddress(), serviceAddress()+":"+port, &adapter.DynamicComponentsConfig{
+		TimeoutInMinutes: 60,
```

Comment:
30 or... 60 is good, too. We'll want for it to be consistent across adapters, though.